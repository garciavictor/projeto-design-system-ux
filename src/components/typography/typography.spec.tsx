import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { H1, H2, H3, Paragraph } from '.';

describe('@components/Typography', () => {
    it('renders H1 correctly', () => {
        const { container } = render(<H1>Heading 1</H1>);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders H2 correctly', () => {
        const { container } = render(<H2>Heading 2</H2>);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders H3 correctly', () => {
        const { container } = render(<H3>Heading 3</H3>);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders Paragraph with default variant correctly', () => {
        const { container } = render(<Paragraph variant="default">Default text</Paragraph>);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders Paragraph with small variant correctly', () => {
        const { container } = render(<Paragraph variant="small">Small text</Paragraph>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
