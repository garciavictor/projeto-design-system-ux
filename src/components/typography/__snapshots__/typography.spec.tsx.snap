// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@components/Typography > renders H1 correctly 1`] = `
<h1
  class="sc-bXdtCk PSZhy"
>
  Heading 1
</h1>
`;

exports[`@components/Typography > renders H2 correctly 1`] = `
<h2
  class="sc-hKDTPf gbwOBZ"
>
  Heading 2
</h2>
`;

exports[`@components/Typography > renders H3 correctly 1`] = `
<h2
  class="sc-fmZSGO gydQkw"
>
  Heading 3
</h2>
`;

exports[`@components/Typography > renders Paragraph with default variant correctly 1`] = `
<p
  class="sc-gwsNht iRelQL"
  variant="default"
>
  Default text
</p>
`;

exports[`@components/Typography > renders Paragraph with small variant correctly 1`] = `
<p
  class="sc-gwsNht cEiqgy"
  variant="small"
>
  Small text
</p>
`;

exports[`Typography Components > renders H1 correctly 1`] = `
<h1
  class="sc-bXdtCk PSZhy"
>
  Heading 1
</h1>
`;

exports[`Typography Components > renders H2 correctly 1`] = `
<h2
  class="sc-hKDTPf gbwOBZ"
>
  Heading 2
</h2>
`;

exports[`Typography Components > renders H3 correctly 1`] = `
<h2
  class="sc-fmZSGO gydQkw"
>
  Heading 3
</h2>
`;

exports[`Typography Components > renders Paragraph with default variant correctly 1`] = `
<p
  class="sc-gwsNht iRelQL"
  variant="default"
>
  Default text
</p>
`;

exports[`Typography Components > renders Paragraph with small variant correctly 1`] = `
<p
  class="sc-gwsNht cEiqgy"
  variant="small"
>
  Small text
</p>
`;
